lib/galaxy/webapps/galaxy/api/histories.py

```diff
@@ -164,41 +192,25 @@ def index(
             description="Whether to return only deleted items.",
             deprecated=True,  # Marked as deprecated as it seems just like '/api/histories/deleted'
         ),
-    ) -> List[AnyHistoryView]:
-        return self.service.index(
-            trans, serialization_params, filter_query_params, deleted_only=deleted, all_histories=all
-        )
-
-    @router.get(
-        "/api/histories/query",
-        summary="Returns histories available to the current user.",
-    )
-    def query(
-        self,
-        response: Response,
-        trans: ProvidesUserContext = DependsOnTrans,
-        limit: Optional[int] = LimitQueryParam,
-        offset: Optional[int] = OffsetQueryParam,
-        show_own: bool = ShowOwnQueryParam,
-        show_published: bool = ShowPublishedQueryParam,
-        show_shared: bool = ShowSharedQueryParam,
-        sort_by: HistorySortByEnum = SortByQueryParam,
-        sort_desc: bool = SortDescQueryParam,
-        search: Optional[str] = SearchQueryParam,
-    ) -> HistoryQueryResultList:
-        payload = HistoryIndexQueryPayload.construct(
-            show_own=show_own,
-            show_published=show_published,
-            show_shared=show_shared,
-            sort_by=sort_by,
-            sort_desc=sort_desc,
-            limit=limit,
-            offset=offset,
-            search=search,
-        )
-        entries, total_matches = self.service.index_query(trans, payload, include_total_count=True)
-        response.headers["total_matches"] = str(total_matches)
-        return entries
+    ) -> Union[List[AnyHistoryView], HistoryQueryResultList]:
```

Comment:
Is there something in `HistoryQueryResultList` that is not covered by `List[AnyHistoryView]`? I would try to converge those if possible too.

Reply:
Adding these optional attributes to the `HistoryDetailed` model sounds good to me :+1: